### PR TITLE
logging integration

### PR DESCRIPTION
**Pull request by supernovahs**

### CODE REVIEW

**Comment by supernovahs** [comment]
leaving logging for bls agg for now . since it will be re implemented a lot anywayy
will do operators info memory after #59  gets merged

**Comment by supernovahs** [comment]
Some logging pending for metrics

**Comment by supernovahs** [comment]
tags criteria for logging
crate-name.file-name.function-name

File: crates/chainio/clients/elcontracts/Cargo.toml
Comment (dralves) [inline crates/chainio/clients/elcontracts/Cargo.toml:18]: alphabetical, please

File: crates/chainio/clients/elcontracts/Cargo.toml
Comment (dralves) [inline crates/chainio/clients/elcontracts/Cargo.toml:27]: alphabetical, please

File: crates/logging/Cargo.toml
Comment (dralves) [inline crates/logging/Cargo.toml:12]: alphabetical, please

File: crates/logging/src/tracing_logger.rs
Comment (dralves) [inline crates/logging/src/tracing_logger.rs:39]: can can you put the Fatal logger at the top? also could you add custom processing so that the Fatal level is indeed fatal (i.e. panics)

File: crates/metrics/Cargo.toml
Comment (dralves) [inline crates/metrics/Cargo.toml:11]: alphabetical, please

File: crates/metrics/collectors/economic/Cargo.toml
Comment (dralves) [inline crates/metrics/collectors/economic/Cargo.toml:11]: alphabetical, please

File: crates/metrics/collectors/economic/src/lib.rs
Comment (dralves) [inline crates/metrics/collectors/economic/src/lib.rs:23]: can you add some comments as to what these metrics are measuring?

File: crates/metrics/collectors/rpc_calls/Cargo.toml
Comment (dralves) [inline crates/metrics/collectors/rpc_calls/Cargo.toml:11]: alphabetical, please

File: crates/metrics/collectors/rpc_calls/src/lib.rs
Comment (dralves) [inline crates/metrics/collectors/rpc_calls/src/lib.rs:28]: can you add some comments as to what these metrics are measuring?

File: crates/metrics/src/eigenmetrics.rs
Comment (dralves) [inline crates/metrics/src/eigenmetrics.rs:22]: can you add some comments as to what these metrics are measuring?

File: crates/metrics/src/eigenmetrics.rs
Comment (dralves) [inline crates/metrics/src/eigenmetrics.rs:21]: does it really need to be called *MetricsMetrics? if yes can you leave an explanation as to why?

File: crates/services/avsregistry/Cargo.toml
Comment (dralves) [inline crates/services/avsregistry/Cargo.toml:19]: alphabetical, please

File: crates/services/avsregistry/src/chaincaller.rs
Comment (dralves) [inline crates/services/avsregistry/src/chaincaller.rs:52]: break the line to 120 chars, please

File: crates/services/avsregistry/src/chaincaller.rs
Comment (dralves) [inline crates/services/avsregistry/src/chaincaller.rs:90]: not valid anymore?

File: crates/services/bls_aggregation/Cargo.toml
Comment (dralves) [inline crates/services/bls_aggregation/Cargo.toml:19]: alphabetical, please

File: crates/services/bls_aggregation/src/bls_agg.rs
Comment (dralves) [inline crates/services/bls_aggregation/src/bls_agg.rs:152]: wrap line to 120 chars, here and elsewhere

File: crates/services/bls_aggregation/src/bls_agg.rs
Comment (dralves) [inline crates/services/bls_aggregation/src/bls_agg.rs:159]: well you're going to crash next, so this is fatal, right?

File: examples/avsregistry-read/Cargo.toml
Comment (dralves) [inline examples/avsregistry-read/Cargo.toml:18]: alphabetical, please

File: examples/info-operator-service/Cargo.toml
Comment (dralves) [inline examples/info-operator-service/Cargo.toml:17]: alphabetical, please

File: crates/metrics/src/eigenmetrics.rs
Comment (supernovahs) [reply crates/metrics/src/eigenmetrics.rs:21]: added an explanation , alternatively we can name this struct as  EigenPerformance also , then the wrapper will be EigenPerformanceMetrics
```
pub struct EigenMetrics {
    /// performance score
    #[metric(
        rename = "eigen_performance_score",
        describe = " A gauge with performance score "
    )]
    performance_score: Gauge,
}
```

File: crates/services/bls_aggregation/src/bls_agg.rs
Comment (supernovahs) [reply crates/services/bls_aggregation/src/bls_agg.rs:179]: https://github.com/Layr-Labs/eigensdk-go/blob/67787e959b727b115628a34e796df3a9ef42f646/services/bls_aggregation/blsagg.go#L247

yup the aggregator will crash

Comment (dralves) [reply crates/services/bls_aggregation/src/bls_agg.rs]: I meant that the log entry should be a fatal not an error

File: crates/metrics/src/eigenmetrics.rs
Comment (dralves) [reply crates/metrics/src/eigenmetrics.rs:21]: let's do it then

**Review by dralves — APPROVED** [review]
only one nit missing, approving assuming it will be addressed

File: crates/services/bls_aggregation/src/bls_agg.rs
Comment (supernovahs) [reply crates/services/bls_aggregation/src/bls_agg.rs]: done

**Pull request merged**
